File: ios/FlatRadar/FlatRadar/Models/AuthModels.swift
import Foundation

// MARK: - Request

struct LoginRequest: Encodable {
    let username: String
    let password: String
    let deviceName: String
    let ttlDays: Int

    enum CodingKeys: String, CodingKey {
        case username, password
        case deviceName = "device_name"
        case ttlDays = "ttl_days"
    }
}

// MARK: - Responses

struct LoginResponse: Decodable {
    let token: String
    let tokenID: Int
    let role: String
    let userID: String?
    let deviceName: String
    let ttlDays: Int

    enum CodingKeys: String, CodingKey {
        case token
        case tokenID = "token_id"
        case role
        case userID = "user_id"
        case deviceName = "device_name"
        case ttlDays = "ttl_days"
    }
}

struct MeResponse: Decodable {
    let role: String
    let userID: String?
    let user: UserInfo?

    enum CodingKeys: String, CodingKey {
        case role
        case userID = "user_id"
        case user
    }
}

File: android/app/src/main/java/com/flatradar/app/domain/model/AuthModels.kt
package com.flatradar.app.domain.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Auth-related DTOs — matches iOS AuthModels.swift.
 */

@JsonClass(generateAdapter = true)
data class LoginRequest(
    val username: String,
    val password: String,
    @Json(name = "device_name") val deviceName: String,
    @Json(name = "ttl_days") val ttlDays: Int = 90
)

@JsonClass(generateAdapter = true)
data class LoginResponse(
    val token: String,
    @Json(name = "token_id") val tokenId: Int? = null,
    val role: String,
    @Json(name = "user_id") val userId: String? = null,
    @Json(name = "device_name") val deviceName: String? = null,
    @Json(name = "ttl_days") val ttlDays: Int? = null,
    val user: UserInfo?
)

@JsonClass(generateAdapter = true)
data class MeResponse(
    val role: String,
    val user: UserInfo?
)

@JsonClass(generateAdapter = true)
data class UserInfo(
    val id: String,
    val name: String,
    val role: String? = null,
    val enabled: Boolean = true,
    @Json(name = "notifications_enabled") val notificationsEnabled: Boolean = false,
    @Json(name = "listing_filter") val listingFilter: ListingFilter? = null
)
